src/Support/Field.php

```diff
@@ -204,26 +206,55 @@ protected function getResolver(): ?Closure
                 }
             }
 
-            // Add the 'selects and relations' feature as 5th arg
-            if (isset($arguments[3])) {
-                $arguments[] = function (int $depth = null) use ($arguments): SelectFields {
-                    $ctx = $arguments[2] ?? null;
-
-                    return new SelectFields($arguments[3], $this->type(), $arguments[1], $depth ?? 5, $ctx);
-                };
-            }
-
             $arguments[1] = $this->getArgs($arguments);
 
             // Authorize
-            if (call_user_func_array($authorize, $arguments) != true) {
+            if (true != call_user_func_array($authorize, $arguments)) {
                 throw new AuthorizationError('Unauthorized');
             }
 
-            return call_user_func_array($resolver, $arguments);
```

Comment:
Why not just calling the app()->call() directly in here? 

```php
return app()->call([ $this, 'resolve' ], [
   'root' => $root,
    'args' => $args, 
    'ctx' => $ctx,
    'resolveInfo' => $arguments[3],
    'selectFields' => $this->instanciateSelectFields($arguments),
]);
```

Reply:
🤷‍♀️

You can always try making a PR!

Reply:
> Would it work?

I can't say, didn't author it /cc @crissi

Reply:
seems like it would bound to specific variable names instead of now where only the 3 first are required arguments and you can use what ever variable name there after as long as the Class name is correct.